### PR TITLE
Allow configuring update_policy

### PR DESCRIPTION
**Pull request by DeviaVir**
First stab at #52

### CODE REVIEW

File: src/spacel/provision/template/app.py [hunk 1]
```diff
@@ -141,6 +143,22 @@ def app(self, app_region):
...
+        # Custom update policy:
+        if app_region.update_policy == 'disabled':
+            del resources['Asg']['UpdatePolicy']
+        if app_region.update_policy == 'redblack':
+            resources['Asg']['UpdatePolicy'] = {
+                'AutoScalingReplacingUpdate': {
+                    'WillReplace': 'true'
+                }
+            }
+            time_pos = (resources['Lc']
```
Comment (thepwagner) [inline src/spacel/provision/template/app.py]: Do we still need this cache-buster?

Given that _everything_ in the `AppManifest` (units, environment variables) is already in UserData, I think redeploys should still trigger correctly without it.

File: src/test/model/test_app.py [hunk 2]
```diff
@@ -50,6 +50,10 @@ def test_valid_invalid_instance_availability(self):
         self.app_region.instance_availability = 'meow'
         self.assertFalse(self.app_region.valid)
 
+    def test_valid_invalid_update_policy(self):
+        self.app_region.update_policy = 'meow'
```
Comment (thepwagner) [inline src/test/model/test_app.py]: Double-noice. 🐱

File: src/spacel/provision/template/app.py (hunk 1 above)
Comment (thepwagner) [inline src/spacel/provision/template/app.py]: I do like the idea of implementing `rotate` within CloudFormation; 'tis the spacel-way: let go, let cloudformation.

I'm not sure about cache-busting 100% of the time though: if I want to add a `sources` entry (i.e. adjust a security group), I shouldn't have to wait ~300s for fresh instances (and pay 2x instance cost for an hour?).

Could we support the "redeploy without changes" case some other (more explicit) way?

**Comment by thepwagner** [comment]
Sanity checking the interplay with `spot`:

We'll adjust the `Asg` resource according to `update_policy`, then ignore whatever `update_policy` is set (by deleting `Asg` and replacing it with `SpotFleet` in the `SpotFleetDecorator`)?

File: src/spacel/provision/template/app.py (hunk 1 above)
Comment (DeviaVir) [inline src/spacel/provision/template/app.py]: Perhaps if we run a `changeset` and don't detect any changes, we can force a rotate?

**Comment by DeviaVir** [comment]
SpotFleet does not currently offer any type of UpdatePolicy, if we stick to cloudformation and hope that AWS will add this at some point later, ignore for now?

File: src/spacel/provision/template/app.py (hunk 1 above)
Comment (DeviaVir) [inline src/spacel/provision/template/app.py]: https://github.com/pebble/spacel-provision/issues/66

**Pull request merged**